deps/rabbit/src/rabbit_queue_location.erl

```diff
@@ -23,26 +32,33 @@
 queue_leader_locators() ->
     ?QUEUE_LEADER_LOCATORS.
 
--spec select_leader_and_followers(amqqueue:amqqueue(), pos_integer()) ->
+-spec select_leader_and_followers(amqqueue:amqqueue(), pos_integer(), atom()) ->
     {Leader :: node(), Followers :: [node()]}.
-select_leader_and_followers(Q, Size)
-  when (?amqqueue_is_quorum(Q) orelse ?amqqueue_is_stream(Q)) andalso is_integer(Size) ->
+select_leader_and_followers(Q, Size, QueueType)
```

Comment:
Argument `QueueType` is redundant.
The queue type is already stored in the `Q` and can be queried using `amqqueue:get_type(Q)` as done in the deleted code.